model/patch_lifecycle.go

```diff
@@ -408,23 +409,42 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 			},
 		)
 	}
-
-	if err = patchVersion.Insert(); err != nil {
-		return nil, errors.WithStack(err)
-	}
-	if err = intermediateProject.Insert(); err != nil {
-		return nil, errors.WithStack(err)
-	}
-	if err = buildsToInsert.InsertMany(ctx, false); err != nil {
-		return nil, errors.Wrapf(err, "error inserting builds for version '%s'", patchVersion.Id)
+	mongoClient := evergreen.GetEnvironment().Client()
+	session, err := mongoClient.StartSession()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to start session")
 	}
-	if err = tasksToInsert.InsertUnordered(ctx); err != nil {
-		return nil, errors.Wrapf(err, "error inserting tasks for version '%s'", patchVersion.Id)
+	defer session.EndSession(ctx)
+
+	txFunc := func(sessCtx mongo.SessionContext) (interface{}, error) {
+		db := evergreen.GetEnvironment().DB()
+		_, err = db.Collection(VersionCollection).InsertOne(ctx, patchVersion)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error inserting version '%s'", patchVersion.Id)
+		}
+		_, err = db.Collection(ParserProjectCollection).InsertOne(ctx, intermediateProject)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error inserting parser project for version '%s'", patchVersion.Id)
+		}
+		if err = buildsToInsert.InsertMany(ctx, false); err != nil {
+			return nil, errors.Wrapf(err, "error inserting builds for version '%s'", patchVersion.Id)
+		}
+		if err = tasksToInsert.InsertUnordered(ctx); err != nil {
+			return nil, errors.Wrapf(err, "error inserting tasks for version '%s'", patchVersion.Id)
+		}
+		if err = p.SetActivated(ctx, patchVersion.Id); err != nil {
```

Comment:
I would move this out of the transaction since it doesn't need to be atomic like everything else. In practice it shouldn't make a difference, but we don't want errors here preventing the rest of the things from inserting